lib/utils/utils.js

```diff
@@ -61,3 +61,27 @@ utils.getPublicMethods = (obj) => {
   });
   return ret;
 };
+
+utils.groupBy = (arr, keyGetter) => {
+  const ret = {};
+  arr.forEach((item) => {
+    const key = keyGetter(item);
+    const group = ret[key];
+    if (!group) {
+      ret[key] = [item];
+    } else {
+      group.push(item);
+    }
+  });
+  return ret;
+};
+
+utils.getGuid = () => {
+  // TODO: make it truly unique
+  function s4() {
+    return Math.floor((1 + Math.random()) * 0x10000)
+      .toString(16)
+      .substring(1);
+  }
+  return `${s4() + s4()}-${s4()}-${s4()}-${s4()}-${s4()}${s4()}${s4()}`;
```

Comment:
This method work, but it may ultimately be more robust to use something like https://www.npmjs.com/package/uuid to generate guids. I see you have a TODO here, so if you want to just leave it as is I can incorporate the uuid package in a separate PR.

Reply:
Yes, it was a temp solution. I'll look into this lib and incorporate it from now.